apps/beeai-ui/src/hooks/useTableSearch.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright 2025 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ChangeEvent, useCallback, useMemo, useState } from 'react';
+
+interface Props<T> {
+  entries: T[];
+  fields: (keyof T)[];
+}
+
+export function useTableSearch<T>({ entries, fields }: Props<T>) {
+  const [search, setSearch] = useState('');
+
+  const searchQuery = search.trim().toLowerCase();
+
+  const onSearch = useCallback((event: ChangeEvent<HTMLInputElement>) => {
+    setSearch(event.target.value);
+  }, []);
+
+  const items = useMemo(() => {
+    if (!searchQuery) {
+      return entries;
+    }
+
+    return entries.filter((item) => fields.some((field) => String(item[field]).toLowerCase().includes(searchQuery)));
+  }, [searchQuery, entries, fields]);
```

Comment:
> In React, the `useMemo` hook will not track a variable that is just a regular local variable (without `useMemo`, `useState`, or `useReducer`) because React's hooks only track **state and props** to determine when to re-run.

Let's change dependency to `search` and calculate `searchQuery` within the `useMemo`.